### PR TITLE
Don't rely on contents of optional Condition fields in CSI mock test

### PR DESCRIPTION
**Pull request by bertinatto**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

In order to promote the volume limits e2e test (from CSI Mock driver) to Conformance, we can't rely  on specific output of optional Condition fields (e.g., Reason and Message). Thus, this PR changes the test to only check the presence of the right condition and verify that the optional fields are not empty.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/storage/csi_mock_volume.go [hunk 1]
```diff
@@ -563,19 +561,15 @@ var _ = utils.SIGDescribe("CSI mock volume", func() {
...
 			return false, err
 		}
-		conditions := pod.Status.Conditions
-		for _, condition := range conditions {
-			matched := reg.MatchString(condition.Message)
-			if condition.Reason == v1.PodReasonUnschedulable && matched {
+		for _, c := range pod.Status.Conditions {
+			// Conformance tests cannot rely on specific output of optional fields (e.g., Reason
+			// and Message) because these fields are not suject to the deprecation policy.
+			if c.Type == v1.PodScheduled && c.Status == v1.ConditionFalse && c.Reason != "" && c.Message != "" {
```
Comment (bertinatto) [inline test/e2e/storage/csi_mock_volume.go]: From https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements:

```
anything that checks optional Condition fields, such as Reason or Message, as these may change over time (however it is reasonable to verify these fields exist or are non-empty) 
```

**Comment by gnufied** [comment]
This will cause false positives where test would appear to pass but may be for some other reason. Can we have 2 versions of this test?

**Comment by msau42** [comment]
> This will cause false positives where test would appear to pass but may be for some other reason. Can we have 2 versions of this test?

We can have another version of the test that's not part of conformance

**Comment by bertinatto** [comment]
> > This will cause false positives where test would appear to pass but may be for some other reason. Can we have 2 versions of this test?
> 
> We can have another version of the test that's not part of conformance

The `volumeLimits` test suite has a test that covers that. That test is even more complete and it uses the csi-hostpath driver instead of the mock driver:

https://github.com/kubernetes/kubernetes/blob/b6b494b4484b51df8dc6b692fab234573da30ab4/test/e2e/storage/testsuites/volumelimits.go#L199-L210

IMO it makes sense to have the less strict mock driver test graduated to _Conformance_  and the stricter _volumeLimits_ testsuite test running in the regular presubmit job.

The only downside I see is that the latter is currently running in a non-required job (**pull-kubernetes-e2e-gce-csi-serial**), but I don't know if that's a big deal.

**Comment by msau42** [comment]
@gnufied what do you think of the hostpath test being more strict than the mock one used for conformance?

**Comment by gnufied** [comment]
yeah that is fine.

/hold cancel
/lgtm

**Pull request merged**